```
______________________________ test_100_continue _______________________________

    def test_100_continue():
        bucket_name = get_new_bucket_name()
        client = get_client()
        client.create_bucket(Bucket=bucket_name)
        objname='testobj'
        resource = '/{bucket}/{obj}'.format(bucket=bucket_name, obj=objname)

        host = get_config_host()
        port = get_config_port()
        is_secure = get_config_is_secure()

        #NOTES: this test needs to be tested when is_secure is True
>       status = _simple_http_req_100_cont(host, port, is_secure, 'PUT', resource)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

s3tests/functional/test_s3.py:6835:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
s3tests/functional/test_s3.py:6804: in _simple_http_req_100_cont
    s = ctx.wrap_socket(s);
        ^^^^^^^^^^^^^^^^^^
/usr/lib64/python3.12/ssl.py:455: in wrap_socket
    return self.sslsocket_class._create(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'ssl.SSLSocket'>
sock = <socket.socket fd=13, family=2, type=1, proto=0, laddr=('0.0.0.0', 0)>
server_side = False, do_handshake_on_connect = True, suppress_ragged_eofs = True
server_hostname = None, context = <ssl.SSLContext object at 0x7f8e7b32d8d0>
session = None

    @classmethod
    def _create(cls, sock, server_side=False, do_handshake_on_connect=True,
                suppress_ragged_eofs=True, server_hostname=None,
                context=None, session=None):
        if sock.getsockopt(SOL_SOCKET, SO_TYPE) != SOCK_STREAM:
            raise NotImplementedError("only stream sockets are supported")
        if server_side:
            if server_hostname:
                raise ValueError("server_hostname can only be specified "
                                 "in client mode")
            if session is not None:
                raise ValueError("session can only be specified in "
                                 "client mode")
        if context.check_hostname and not server_hostname:
>           raise ValueError("check_hostname requires server_hostname")
E           ValueError: check_hostname requires server_hostname

/usr/lib64/python3.12/ssl.py:969: ValueError
```

--- a/s3tests/functional/test_s3.py
+++ b/s3tests/functional/test_s3.py
@@ -6800,7 +6800,8 @@ def _simple_http_req_100_cont(host, port, is_secure, method, resource):
 
     s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     if is_secure:
-        s = ssl.wrap_socket(s);
+        ctx = ssl.create_default_context()
+        s = ctx.wrap_socket(s, server_hostname=host);
     s.settimeout(5)
     s.connect((host, port))
     s.send(req)

